src/datapoints-graph/charts/chart-realtime.service.ts

```diff
@@ -149,6 +222,58 @@ export class ChartRealtimeService {
       seriesMatchingDatapoint.data = this.removeValuesBeforeTimeRange(
         seriesMatchingDatapoint
       );
+
+      alarmsAndEvents.forEach((item) => {
+        if (
+          typeof seriesMatchingDatapoint.data !== 'object' ||
+          seriesMatchingDatapoint.data === null
+        ) {
+          return;
+        }
+        const renderType: DatapointChartRenderType =
+          datapoint.renderType || 'min';
+        const dp: DatapointWithValues = {
+          ...datapoint,
+          values: seriesMatchingDatapoint.data as {
+            [date: string]: { min: number; max: number }[];
+          },
+        };
+
+        if (isEvent(item)) {
+          const newEventSeries =
+            this.echartsOptionsService.getAlarmOrEventSeries(
+              dp,
+              renderType,
+              false,
+              [item],
+              'event',
+              item.id
+            );
+          allDataSeries.push(...newEventSeries);
+        } else if (isAlarm(item)) {
+          const alarmExists = allDataSeries.some((series: { data: any[] }) =>
+            series.data.some((data) => data[0] === item.creationTime)
+          );
+          if (alarmExists) {
+            const alarmSeries = allDataSeries.find((series: { data: any[] }) =>
+              series.data.some((data) => data[0] === item.creationTime)
```

Comment:
Why not `id` instead of `creationTime`?

Reply:
data[0] is the creationTime of an existing alarm series so that's how we check if they already exist, as this is the only property which does not change.